From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrea Righi <andrea.righi@canonical.com>
Date: Wed, 6 Mar 2024 19:49:15 +0100
Subject: [PATCH 90/90] UBUNTU: SAUCE: apparmor4.0.0: LSM stacking v39: fix
 build error with CONFIG_SECURITY=n

When CONFIG_SECURITY=n we are not initializing lsmblob correctly in
security_cred_getlsmblob().

Fix by calling lsmblob_init() properly.

Fixes: 4d652c1a90a7 ("UBUNTU: SAUCE: apparmor4.0.0 [13/87]: LSM stacking v39: LSM: Create new security_cred_getlsmblob LSM hook")
Reported-by: Emil Renner Berthing <emil.renner.berthing@canonical.com>
Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 include/linux/security.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/security.h b/include/linux/security.h
index b32c24425e40..6432d00aab29 100644
--- a/include/linux/security.h
+++ b/include/linux/security.h
@@ -1170,7 +1170,7 @@ static inline void security_cred_getsecid(const struct cred *c, u32 *secid)
 static inline void security_cred_getlsmblob(const struct cred *c,
 					    struct lsmblob *blob)
 {
-	*secid = 0;
+	lsmblob_init(blob);
 }
 
 static inline int security_kernel_act_as(struct cred *cred, u32 secid)
